src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -139749,6 +139909,178 @@ void M()
+    {
+        const bool b = true;
+        s1 = null;
+        return b; // 1
+    }
+
+    static bool M2([MaybeNullWhen(false)] out string s1)
+    {
+        s1 = null;
+        return (bool)true; // 2
+    }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullWhenAttributeDefinition });
+            comp.VerifyEmitDiagnostics(
+                // (12,9): warning CS8762: Parameter 's1' must have a non-null value when exiting with 'true'.
+                //         return b; // 1
+                Diagnostic(ErrorCode.WRN_ParameterConditionallyDisallowsNull, "return b;").WithArguments("s1", "true").WithLocation(12, 9),
+                // (18,9): warning CS8762: Parameter 's1' must have a non-null value when exiting with 'true'.
+                //         return (bool)true; // 2
+                Diagnostic(ErrorCode.WRN_ParameterConditionallyDisallowsNull, "return (bool)true;").WithArguments("s1", "true").WithLocation(18, 9)
+                );
+        }
+
+        [Fact, WorkItem(48126, "https://github.com/dotnet/roslyn/issues/48126")]
+        public void EnforcedInMethodBody_MaybeNullWhen_Issue_48126_3()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+class C
+{
+    static bool M1([MaybeNullWhen(false)] out string s1)
+    {
+        const bool b = true;
+        s1 = null;
+        return b; // 1
+    }
+
+    static bool M2([MaybeNullWhen(false)] out string s1)
+    {
+        const bool b = true;
+        s1 = null;
+        return b && true; // 2
+    }
+
+    static bool M3([MaybeNullWhen(false)] out string s1)
+    {
+        const bool b = false;
+        s1 = null;
+        return b;
+    }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullWhenAttributeDefinition });
+            comp.VerifyEmitDiagnostics(
+                // (12,9): warning CS8762: Parameter 's1' must have a non-null value when exiting with 'true'.
+                //         return b; // 1
+                Diagnostic(ErrorCode.WRN_ParameterConditionallyDisallowsNull, "return b;").WithArguments("s1", "true").WithLocation(12, 9),
+                // (19,9): warning CS8762: Parameter 's1' must have a non-null value when exiting with 'true'.
+                //         return b && true; // 2
+                Diagnostic(ErrorCode.WRN_ParameterConditionallyDisallowsNull, "return b && true;").WithArguments("s1", "true").WithLocation(19, 9)
+                );
+        }
+
+        [Fact, WorkItem(48126, "https://github.com/dotnet/roslyn/issues/48126")]
+        public void EnforcedInMethodBody_MaybeNullWhen_Issue_48126_4()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+class C
+{
+    static bool M1([MaybeNullWhen(false)] out string s1)
+    {
+        return M1(out s1);
+    }
+
+    static bool M2([MaybeNullWhen(false)] out string s1)
+    {
+        return !M1(out s1); // 1
```

Comment:
Sorry if I missed it--do we have a test where negation of the method call is *required* to avoid the "Parameter 's1' must have a non-null value when exiting with 'true'" warning? e.g. a test similar to this one except `M1` has `MaybeNullWhen(true)`.